packages/web-components/fast-foundation/src/combobox/combobox.ts

```diff
@@ -0,0 +1,602 @@
+            return true;
+        }
+
+        switch (key) {
+            case "Enter": {
+                this.updateValue(true);
+                this.open = false;
+                this.control.setSelectionRange(
+                    this.control.value.length,
+                    this.control.value.length
+                );
+                break;
+            }
+
+            case "Escape": {
+                if (this.open) {
+                    this.open = false;
+                    this.filter = this.control.value;
+                    this.filterOptions();
+                } else {
+                    this.value = "";
+                    this.control.value = "";
+                }
+                break;
+            }
+
+            case "Tab": {
+                if (!this.open) {
+                    return true;
+                }
+
+                e.preventDefault();
+                this.open = false;
+                break;
+            }
+
+            case "ArrowUp":
+            case "ArrowDown": {
+                if (!this.open) {
+                    this.open = true;
+                    return true;
+                }
+
+                this.filterOptions();
+
+                if (this.options.length > 0) {
+                    super.keydownHandler(e);
+                }
+
+                if (this.autocomplete === "inline" || this.autocomplete === "both") {
+                    this.setInlineSelection();
+                    this.updateValue();
+                }
+
+                break;
+            }
+
+            default: {
+                return true;
+            }
+        }
+    }
+
+    /**
+     * Handle keyup actions for value input and text field manipulations.
+     *
+     * @param e - the keyboard event
+     * @internal
+     */
+    public keyupHandler(e: KeyboardEvent): boolean | void {
+        const key = e.key;
+
+        if (key.length === 1) {
+            this.filter += e.key;
+        }
+
+        switch (key) {
+            case "Backspace": {
+                this.filter = this.control.value;
+                this.filterOptions();
+                this.selectedIndex = -1;
+                this.setSelectedOptions();
+                break;
+            }
+
+            case "ArrowLeft":
+            case "ArrowRight":
+            case "Home":
+            case "End": {
+                if (this.autocomplete === "both") {
+                    this.filter = this.control.value;
+                    this.filterOptions();
+                } else {
+                    this.selectedIndex = -1;
+                    this.setSelectedOptions();
+                }
+                break;
+            }
+
+            default: {
+                if (key.length === 1 && key.match(/\S/)) {
+                    this.filterOptions();
+                    if (this.autocomplete === "list" || this.autocomplete === "both") {
+                        if (!this.open && this.control.value.length) {
```

Comment:
```suggestion
Some of these if statements can be consolidated since they only nest. Example:

```suggestion
                    if (
                        (this.autocomplete === "list" || this.autocomplete === "both") &&
                        (!this.open && this.control.value.length)
                    ) {
```
```